staging/src/k8s.io/sample-controller/artifacts/examples/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//This is an example of polymorphic scale client.
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/clientcmd"
+	clientset "k8s.io/sample-controller/pkg/client/clientset/versioned"
+)
+
+const (
+	Namespace       = "default"
+	ResourceName    = "example-foo"
+	ResourceGroup   = "samplecontroller.k8s.io"
+	ResourceVersion = "v1alpha1"
+)
+
+func main() {
```

Comment:
I was thinking that it would be better to integrate with the existing controller.

Something like scaling Foo so that it increases `.spec.Replicas` and hence scaling the deployment as well.

Reply:
Had a discussion with @wanghaoran1988 on IRC and:

- We can include this as a separate example like in this PR.
- But we should mention that users can use `kubectl scale --replicas=n foo/name` to scale Foo and hence the deployment.